tests/highlevel/test_messages.py

```diff
@@ -944,19 +923,67 @@ def test_initial(self):
         self.PYDEVD_CMD = CMD_SET_BREAK
         self.assert_received(self.debugger, [
             self.expected_pydevd_request(
-                '1\tpython-line\tspam.py\t10\tNone\tNone\tNone'),
+                '1\tpython-line\tspam.py\t10\tNone\tNone\tNone\tNone'),
+            self.expected_pydevd_request(
+                '2\tpython-line\tspam.py\t15\tNone\ti == 3\tNone\tNone'),
+        ])
+
+    def test_with_hit_condition(self):
+        with self.launched():
+            self.send_request(
+                source={'path': 'spam.py'},
+                breakpoints=[
+                    {'line': '10',
+                     'hitCondition': '5'},
+                    {'line': '15',
+                     'hitCondition': ' ==5'},
+                    {'line': '20',
+                     'hitCondition': '> 5'},
+                    {'line': '25',
+                     'hitCondition': '% 5'},
+                ],
+            )
+            received = self.vsc.received
+
+        self.assert_vsc_received(received, [
+            self.expected_response(
+                breakpoints=[
+                    {'id': 1,
+                     'verified': True,
+                     'line': '10'},
+                    {'id': 2,
+                     'verified': True,
+                     'line': '15'},
+                    {'id': 3,
+                     'verified': True,
+                     'line': '20'},
+                    {'id': 4,
+                     'verified': True,
+                     'line': '25'},
+                ],
+            ),
+            # no events
+        ])
+        self.PYDEVD_CMD = CMD_SET_BREAK
+        self.assert_received(self.debugger, [
+            self.expected_pydevd_request(
+                '1\tpython-line\tspam.py\t10\tNone\tNone\tNone\t@HIT@ == 5'),
+            self.expected_pydevd_request(
+                '2\tpython-line\tspam.py\t15\tNone\tNone\tNone\t@HIT@ ==5'),
+            self.expected_pydevd_request(
+                '3\tpython-line\tspam.py\t20\tNone\tNone\tNone\t@HIT@ > 5'),
             self.expected_pydevd_request(
-                '2\tpython-line\tspam.py\t15\tNone\ti == 3\tNone'),
+             '4\tpython-line\tspam.py\t25\tNone\tNone\tNone\t@HIT@ % 5 == 0'),
         ])
 
     def test_with_existing(self):
         with self.launched():
             with self.hidden():
                 self.PYDEVD_CMD = CMD_SET_BREAK
                 self.expected_pydevd_request(
-                    '1\tpython-line\tspam.py\t10\tNone\tNone\tNone')
+                    '1\tpython-line\tspam.py\t10\tNone\tNone\tNone\tNone')
```

Comment:
This is a non-backwards-compatible pydevd protocol change, which might make it harder to upstream. I think we shouldn't  be reporting this field if it's `None` - that way old breakpoints works as they always did, and only the new ones get the extra new field to parse.

Reply:
In pydevd, I support the old and the new breakpoints. However, should I just add CMD_SET_BREAK2 that can be configured using json?

Reply:
Oh, sorry, I got confused about the direction here. Never mind!